src/components/shared/thread/ActivityGraphFills.js

```diff
@@ -410,52 +411,41 @@ export class ActivityGraphFillComputer {
     const afterSampleCpuRatio =
       cpuAfterSample === null ? 1 : cpuAfterSample / maxThreadCPUDelta;
 
-    // Samples have two parts to be able to present the CPU utilizations properly.
-    // The first half of the sample will use the CPU delta number that belongs to
-    // this sample.
+    // Samples have two parts to be able to present the different CPU usages properly.
+    // This is because CPU usage number of a sample represents the CPU usage
+    // starting starting from the previous sample time to this sample time.
+    // These parts will be:
+    // - Between `sampleCategoryStartPixel` and `samplePixel` with beforeSampleCpuRatio.
+    // - Between `samplePixel` and `sampleCategoryEndPixel` with afterSampleCpuRatio.
+
+    // Here we are accumulating the first part of the sample. It will use the
+    // CPU delta number that belongs to this sample.
+    // This part starts from the "sample start time" to "sample time" and uses
+    // beforeSampleCpuRatio.
     for (let i = intCategoryStartPixel; i <= intSamplePixel; i++) {
       percentageBuffer[i] += beforeSampleCpuRatio;
     }
-    // The second half of the sample will use the CPU delta number that belongs to
-    // the next sample.
-    // For the samples that are consist of only one pixel, this loop will not be
-    // executed. It will only be executed if sample has 2 or more pixels.
-    for (let i = intSamplePixel + 1; i <= intCategoryEndPixel; i++) {
-      percentageBuffer[i] += afterSampleCpuRatio;
-    }
-
-    // If a sample is only one pixel, then only the first for loop in is being
-    // run. If a sample has more than one pixel, then both the first and the
-    // second loop is being run. If there is only one pixel, which is
-    // intCategoryStartPixel === intCategoryEndPixel case, we should use the
-    // first half ratio to compute the sub pixel subtraction. This is because we
-    // only use the first half ratio to compute in that case. If there are 2 or
-    // more pixels, then we use the second half ratio as the ending.
-    let sampleEndRatio;
-    if (intCategoryStartPixel === intCategoryEndPixel) {
-      // Sample has only one pixel in the activity graph. Therefore use the
-      // first half ratio as the end ratio.
-      sampleEndRatio = beforeSampleCpuRatio;
-    } else {
-      // Sample has more than one pixel in the activity graph. Therefore use the
-      // second half ratio as the end ratio.
-      sampleEndRatio = afterSampleCpuRatio;
-    }
```

Comment:
The old implementation was wrong and also unnecessarily more complex. The new implementation became a lot simpler after handling the before and after parts' own subpixel subtraction.